### PR TITLE
MAINT: 1.8.1 backports/prep

### PR DESCRIPTION
**Pull request by tylerjereddy**
* backport 10/11 of the current `backport-candidate` PRs--one is missing because https://github.com/scipy/scipy/pull/15561#issuecomment-1086747474 is a tad annoying to do manually
* probably the most important thing getting fixed is the restoration of Pythran for Windows builds?
* unfortunately, I don't see any signs that we'll have the `PROPACK` stuff fully enabled/Windows portable for `1.8.1` (some issues came up last minute for `1.8.0`, so have an environment variable to control its usage on the maintenance branch)
* this is the first test of the new authors machinery updates from @tupui

### CODE REVIEW

File: doc/release/1.8.1-notes.rst
Comment (tylerjereddy) [inline doc/release/1.8.1-notes.rst:14]: this author was added manually for critical contributions fixing the `.htaccess` doc server file following the last release with new doc theme

**Comment by tupui** [comment]
Would it be possible to add #15380 to the list? This way the version switcher would be in both dev and released versions and we can think about deprecating the webpage listing the docs. (Then it needs an entry for 1.8.1 in the json).

**Comment by rgommers** [comment]
Diff and included backports here seem fine to me; there's a bug in here somewhere though since CI is failing with:
```
ImportError: cannot import name '_broadcast_arrays' from 'scipy.stats._axis_nan_policy'
```
 
> Would it be possible to add #15380 to the list? This way the version switcher would be in both dev and released versions and we can think about deprecating the webpage listing the docs. (Then it needs an entry for 1.8.1 in the json).

That's a new docs feature, quite large and changes a doc build dependency. It's okay I guess, but I'd personally be inclined to leave it for 1.9.0. Up to @tylerjereddy though.

**Comment by tylerjereddy** [comment]
Two updates here:

1. for the CI failure, I've pushed in a commit that allows tests to pass locally on this branch--``_broadcast_arrays()` function lives somewhere else on `main`, but in this branch the import can be deleted because it resides directly in `_hypotests.py`.
2. For the HiGHS segfault backport--there's nothing to backport, the segfault problem and related files don't even exist on this branch--see analysis starting here: https://github.com/scipy/scipy/pull/15561#issuecomment-1100957547

**Comment by tylerjereddy** [comment]
Ah, I feel like I've heard a few discussions related to the CI failure with `ImportError: cannot import name 'contextfunction' from 'jinja2'`.

**Comment by rgommers** [comment]
> Edit: I see a CI `Lint` failure for a long regex from Pamphile's new author script I think--not sure it is really worth trying to split that over multiple lines to satisfy the check.

Yes, best to ignore I'd say.

> Ah, I feel like I've heard a few discussions related to the CI failure with `ImportError: cannot import name 'contextfunction' from 'jinja2'`.

Yes, I think you need the pin as in gh-15895. Better to put an upper bound in `jinja` than update Sphinx version in a maintenance branch probably.

**Comment by tylerjereddy** [comment]
Ok, let's see if pinning `Jinja2` and `Pillow` versions will fix the respective doc-related issues for Azure and CircleCI.

**Comment by tylerjereddy** [comment]
CI a bit cleaner now--the benchmark failure is a timeout--I believe we've had some activity surrounding that on `main` as well.

**Comment by tupui** [comment]
Yes there was the PDF removal in #15649 and move to Meson in #15686

**Review by rgommers — APPROVED** [review]
> For the benchmark timeout I had it until I fixed the env variable `OPENBLAS_NUM_THREADS=1`

It'd be nice to add that here now that we identified the issue, to prevent seeing this on future PRs to this branch. If it doesn't work at once, let's ignore it. This PR is good to go.

**Comment by tylerjereddy** [comment]
I think bleeding edge NumPy is causing a few deprecation issues for matrix subclasses and distutils. Worth suppressing in our suite?

**Comment by tupui** [comment]
> I think bleeding edge NumPy is causing a few deprecation issues for matrix subclasses and distutils. Worth suppressing in our (maintenance branch) suite?

Since it's just impacting the test suite I think we can skip the matrix issue.

**Comment by tylerjereddy** [comment]
* the benchmark timeout persists in CI, but sounds like reviewers are "ok" with ignoring that since the obvious fix didn't help enough
* folks don't seem too worried about bleeding-edge NumPy causing 4 test suite errors--1 is precision-related, the others are deprecations--I'll go ahead and merge this then--if an urgent concern pops up regarding that let me know, my next step will be a test PR in the wheels repo to reactivate Pythran on Windows to see if all is well there (so, probably another day or so for last minute adjustments even if I don't see more problems over there).

**Pull request merged**
